File: abc/abc123/abc123_c.cs
// https://atcoder.jp/contests/abc123/submissions/8450275

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        public static void Main()
        {
            var N = long.Parse(Console.ReadLine());
            var city = new long[5].Select(_ => long.Parse(Console.ReadLine())).ToArray();

            Console.WriteLine((long)Math.Ceiling((double)N / city.Min()) + 4);
        }
    }
}

File: abc/abc123/abc123_c.fs
// https://atcoder.jp/contests/abc123/submissions/12198540

let N = stdin.ReadLine() |> float

let cities = Array.zeroCreate<float> 5

for i in [ 0..4 ] do
    cities.[i] <- stdin.ReadLine() |> float

N / (Array.min cities) + 4.
|> ceil
|> int64
|> stdout.WriteLine
